packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1415,6 +1416,12 @@ export class ContainerRuntime
 		this.submitSummaryFn = submitSummaryFn;
 		this.submitSignalFn = submitSignalFn;
 
+		if (isInstanceOfISnapshot(snapshotWithContents)) {
+			this.evaluateSnapshotForOmittedBlobContents(
+				snapshotWithContents.snapshotTree,
+				snapshotWithContents.blobContents,
+			);
```

Comment:
Why do we need it here? I.e. why this is not a call at data store level?

Reply:
Basically there is only one place that needs it - setting this.snapshotFetchRequired in RemoteFluidDataStoreContext constructor.
There is another usage in FluidDataStoreContext.isRoot(), but we can either overwrite isRoot() in RemoteFluidDataStoreContext, or promote this.snapshotFetchRequired to FluidDataStoreContext